### DISCUSSION TITLE
Nested objects in isar?

### DISCUSSION THREAD

**Discussion by davoutuk** [isar/isar#343] [Help]
I am building a card game app in Flutter,  that will include two objects:   'Game' and 'Hand'

The 'Hand' object is set as a nested array within the 'Game' object,  and a Hand instance is only ever linked to a single Game.

What's the best approach for persisting wholly owned nested objects?

**Comment by simc** [comment]
That's not possible currently unfortunately. Do you have examples of other databases that support this mechanism?

I had a hard time figuring out how to query nested objects so I didn't want to rush adding this capability.

**Reply by davoutuk** [reply]
I've no requirement to query the nested objects.   Instead I just want a method to persist the main and nested objects in one call.   Is that possible?

**Reply by simc** [reply]
Currently only using [TypeConverters](https://isar.dev/type_converters.html) but it will be possible in the future. I have no estimate however when it will be possible.

**Comment by outlaws42** [comment]
MongoDB can nest documents inside of documents. I am wondering how to do something similar. Currently, i have connected my flutter app to an API backend that I created with python and MongoDB.  I wanted to make a version that had a local DB only and maybe do a sync type of situation eventually. It is a score keeping app that I create a match with a game and list of players that are playing that game. In mongo, I nest the list of players inside the match document.  Is this still not possible in Isar?

**Reply by simc** [reply]
This is possible in Isar 3.0

Embedded objects are now supported. Check out the docs on isar.dev

**Reply by outlaws42** [reply]
Cool, thank you!!
